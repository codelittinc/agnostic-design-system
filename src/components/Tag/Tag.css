:root {
  --tag-container-background-color: var(--structure-color-100);
  --tag-container-border-radius: 4px;
  --tag-container-color: var(--structure-color-600);
  --tag-container-padding: 4px;
  --tag-container-width: fit-content;
}

.tag-container {
  background-color: var(--tag-container-background-color);
  border-radius: var(--tag-container-border-radius);
  color: var(--tag-container-color);
  font-family: var(--tag-container-font-family);
  font-size: var(--tag-container-font-size);
  font-weight: var(--tag-container-font-weight);
  margin: auto;
  padding: var(--tag-container-padding);
  width: var(--tag-container-width);
}
