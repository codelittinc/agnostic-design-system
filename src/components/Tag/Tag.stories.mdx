# Tag Component

## Type: Simple

## Tag Props

|        Prop        |      Type       | Default | Mandatory |
| :----------------: | :-------------: | :-----: | :-------: |
|      children      | React.ReactNode |    -    |   true    |
| variablesClassName |     string      |    -    |   false   |

## Getting Started

```javascript
import React from 'react';
import { Tag } from '@codelittinc/agnostic-design-system';

const MyComponent = () => {
  return (
    <div>
      <Tag>Tag Name</Tag>
    </div>
  );
};

export default MyComponent;
```

## CSS Variables Available

| Property  |    Attribute     | State |
| :-------: | :--------------: | :---: |
| container | background-color |   -   |
| container |  border-radius   |   -   |
| container |      color       |   -   |
| container |   font-family    |   -   |
| container |    font-size     |   -   |
| container |   font-weight    |   -   |
| container |      width       |   -   |
