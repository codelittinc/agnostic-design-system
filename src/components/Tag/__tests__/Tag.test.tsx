import React from 'react';
import Tag from '@/components/Tag';
import { render } from '@testing-library/react';

describe('Tag', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Tag>Testing Tag</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });
});
