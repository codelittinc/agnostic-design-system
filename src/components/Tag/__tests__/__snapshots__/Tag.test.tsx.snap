// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<DocumentFragment>
  <div
    class="tag-container"
  >
    Testing Tag
  </div>
</DocumentFragment>
`;
